optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -160,9 +167,24 @@ public final Solution_ solve(Solution_ problem) {
         boolean restartSolver = true;
         while (restartSolver) {
             solvingStarted(solverScope);
-            runPhases(solverScope);
-            solvingEnded(solverScope);
-            restartSolver = checkProblemFactChanges();
+            LongTaskTimer.Sample sample = solveLengthTimer.start();
+            try {
+                runPhases(solverScope);
+
+                solvingEnded(solverScope);
+                sample.stop();
+
+                restartSolver = checkProblemFactChanges();
+            } catch (Exception e) {
+                // Can't store this in a local field; don't know in advance
+                // if we'll have an exception, and what kind of exception
+                // it is
+                sample.stop();
+                Metrics.counter("optaplanner.solver.errors",
+                        "exceptionClass", e.getClass().getSimpleName(),
+                        "message", e.getMessage());
```

Comment:
First of all, does this actually measure anything? I see the counter being created, but not used.
Second, do we need to tag by exception and message? Is that important information?

I think we need just error count. And in that case, you can keep it in a field.